kotlin-native/backend.native/tests/runtime/program_name/runtime_program_name.kt

```diff
@@ -0,0 +1,11 @@
+@file:OptIn(kotlin.experimental.ExperimentalNativeApi::class)
+
+import kotlin.test.*
+import kotlin.native.Platform
+
+fun main(args: Array<String>) {
+    // Remove path and extension (.kexe or .exe)
+    val programFileName = Platform.programName.substringAfterLast("/").substringBeforeLast(".")
+
+    assertEquals("program_name", programFileName)
+}
```

Comment:
More tests are necessary. For example, the original issue mentions a nice use case:
> Build multi-call binaries. For example, busybox is a single binary that contains many tools, and decides which tool to run based on how the symbolic link is called: https://busybox.net/downloads/BusyBox.html#usage (that's how it can be so small)

So checking a case with symbolic link would also be useful.

Reply:
I added test for that by calling `execv()` in the C code. This allowed to test even more use-cases (no programName at all), while still covering the use-case of renaming/linking a binary.